rcl/src/rcl/network_flow_endpoints.c

```diff
@@ -0,0 +1,122 @@
+// Copyright 2020 Ericsson AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/error_handling.h"
+#include "rcl/graph.h"
+#include "rcl/network_flow_endpoints.h"
+#include "rcl/publisher.h"
+#include "rcl/subscription.h"
+
+#include "rcutils/allocator.h"
+#include "rcutils/macros.h"
+#include "rcutils/types.h"
+
+#include "rmw/error_handling.h"
+#include "rmw/get_network_flow_endpoints.h"
+#include "rmw/network_flow_endpoint_array.h"
+#include "rmw/types.h"
+
+#include "./common.h"
+
+rcl_ret_t
+__validate_allocator_and_network_flow_endpoint_array(
+  rcutils_allocator_t * allocator,
```

Comment:
The allocator seems unused here. Maybe it shouldn't be passed?

Reply:
Or, I see that is just combining the validation of these two things, which is ok I guess, but they could just be separate functions, or you could just include the validation of the allocator inplace, since it's only a single line...

Reply:
Fixed. See [here](https://github.com/ros2/rcl/pull/880/commits/8b4f47ed989ac56c89f07bd4cee3d56a8c315a14).